### ISSUE TITLE
cpp_utils missing

### ISSUE THREAD

**Issue by steven-varga** [wichtounet/etl#3]
The library seem to be missing some parts:
/usr/local/include/etl/std.hpp:21:32: fatal error: cpp_utils/compat.hpp: No such file or directory
from "/etl/std.hpp" the inclusions are not provided
....
// cpp_utils
#include "cpp_utils/compat.hpp"
#include "cpp_utils/tmp.hpp"
#include "cpp_utils/likely.hpp"
#include "cpp_utils/assert.hpp"
#include "cpp_utils/parallel.hpp"

investingating source tree, no files in git directory. 

Method of installation:
git clone https://github.com/wichtounet/etl.git
cp -R etl /usr/local/lib

**Comment by wichtounet** [comment]
Hello, 

cpp_utils is  a sub module, therefore, you need to do a recursive clone: 

git clone --recursive https://github.com/wichtounet/etl.git

I also recommend only copying the headers: 

cp -R etl/include /usr/local/lib/include/
cp -R lib/include /usr/local/lib/include/

**Comment by josephjaspers** [comment]
used git clone --recursive, same issue.

**Comment by wichtounet** [comment]
Hi @josephjaspers 

Sorry about that. 

Do you a lib/include/cpp_utils directory in the directory you clone with --recursive ?

Can you give me the command you are using for compiling and the error ? 

Thanks

**Comment by josephjaspers** [comment]
I simply used:

git clone --recursive
found the issue, you must include both the:

 etl/include
AND
etl/lib/include

This seems counter-intuitive to a header-library. Why not put the etl/lib directory into the etl/lib directory so users won't need to include these two directories separately?

**Comment by wichtounet** [comment]
You are right, you need both. I forgot to mention this in the Readme. I fixed it. Thanks :)

I didn't want to fix the dependencies of the library with the library. But this may not be ideal indeed as you pointed out.
